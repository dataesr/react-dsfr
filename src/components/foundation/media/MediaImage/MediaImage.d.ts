import * as React from 'react';

export type MediaImageSize = 'sm' | 'md' | 'lg';

export interface MediaImageProps {
  alt?: string;
  className?: string;
  legend?: string;
  size?: MediaImageSize;
  src: string;
}

declare const MediaImage: React.FC<MediaImageProps>;

export default MediaImage;
