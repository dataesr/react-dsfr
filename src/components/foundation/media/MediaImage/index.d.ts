import MediaImage from "./MediaImage"

export { MediaImage }
