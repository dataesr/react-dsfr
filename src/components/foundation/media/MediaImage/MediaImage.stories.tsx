import { ComponentStory } from '@storybook/react';

import MediaImage from './MediaImage';

// https://www.systeme-de-design.gouv.fr/elements-d-interface/fondamentaux-techniques/medias

export default {
  title: 'Fondamentaux Techniques/Médias Image - Media Image',
  component: MediaImage,
  argTypes: {
    size: {
      description: `Si vous souhaitez accentuer la visibilité d'une image vous pouvez utiliser les différentes tailles à votre disposition :
      - lg : dans cette taille, la largeur de l'image est supérieure à celle du contenu.
      - md : dans cette taille, la largeur de l'image correspond à celle du contenu. Par défaut (100%)
      - sm : dans cette taille, la largeur de l'image est inférieure à celle du contenu.
      `,
    },
  },
  parameters: {
    docs: {
      description: {
        component: 'Les médias sont des images ou des vidéos présents au sein des pages du site. Ils sont utilisés pour donner une information visuelle ou dans un but purement décoratif.',
      },
    },
  },
};

const Template: ComponentStory<typeof MediaImage> = (args) => <MediaImage {...args} />;

export const Image = Template.bind({});
Image.args = {
  src: 'https://3.bp.blogspot.com/-PyzgZNWTD7c/UbzG65ExiMI/AAAAAAAABlw/ODiasL9Nb0Y/s320/ASPIRE+Wallpaper.jpg',
  alt: 'alternative description',
  legend: 'legend',
  size: 'md',
};

Image.storyName = 'Image';
