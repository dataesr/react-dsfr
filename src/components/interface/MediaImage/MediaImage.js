import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

/**
 *
 * @visibleName MediaImage
 */
const MediaImage = ({
  alt,
  className,
  legend,
  size,
  src,
}) => {
  const _className = classNames(
    'fr-content-media',
    `fr-content-media--${size}`,
    className,
  );

  return (
    <figure
      className={_className}
      data-testid="image"
    >
      <div className="fr-content-media__img">
        <img src={src} alt={alt} />
      </div>
      <figcaption className="fr-content-media__caption">{legend}</figcaption>
    </figure>
  );
};

MediaImage.defaultProps = {
  alt: '',
  className: '',
  legend: '',
  size: 'md',
};

MediaImage.propTypes = {
  alt: PropTypes.string,
  className: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object,
    PropTypes.array,
  ]),
  legend: PropTypes.string,
  size: PropTypes.oneOf(['sm', 'md', 'lg']),
  src: PropTypes.string.isRequired,
};

export default MediaImage;
