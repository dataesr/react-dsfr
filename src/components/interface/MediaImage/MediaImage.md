## Médias (intégrés au contenu) - Responsive medias
Les médias désignent vos contenus photos et vidéos. Lorsqu’ils sont intégré à une page de contenu, il est recommandé de suivre les règles décrites ci-dessous.

### Image intégrée dans un contenu
Lorsqu’une image est intégrée dans un contenu éditorial, elle est présentée en format 16:9 et est précédée et suivie d'un espacement, et d'une légende comme illustré ci-dessous.
```jsx
  <MediaImage
    src="https://bibliotheques.csdm.qc.ca/files/2018/11/10_banques_dimages_gratuites_libres_de_droits-300x169.jpg"
    alt="description alternative à l‘image"
    legend="légende de l'image"
  />
```

#### Les trois tailles proposées pour les images
Si vous souhaitez accentuer la visibilité d'une image vous pouvez utiliser les différentes tailles à votre disposition :
- LG : dans cette taille, la largeur de l’image est supérieure à celle du contenu.
- MD : dans cette taille, la largeur de l’image correspond à celle du contenu. Par défaut (100%)
- SM : dans cette taille, la largeur de l’image est inférieure à celle du contenu.

#### Image taille LG (125%)
L’image est plus grande que la zone de texte.
```jsx
  <MediaImage
    src="https://bibliotheques.csdm.qc.ca/files/2018/11/10_banques_dimages_gratuites_libres_de_droits-300x169.jpg"
    alt="description alternative à l‘image"
    legend="légende de l'image"
    size="lg"
  />
```

#### Image taille SM (75%)
L’image est plus petite que la zone de texte.
```jsx
  <MediaImage
    src="https://bibliotheques.csdm.qc.ca/files/2018/11/10_banques_dimages_gratuites_libres_de_droits-300x169.jpg"
    alt="description alternative à l‘image"
    legend="légende de l'image"
    size="sm"
  />
```