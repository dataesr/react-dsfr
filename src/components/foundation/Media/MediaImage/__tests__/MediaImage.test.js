import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import MediaImage from '../index';

describe('<MediaImage />', () => {
  it('renders correctly', () => {
    const component = renderer
      .create(
        <MediaImage
          src="https://bibliotheques.csdm.qc.ca/files/2018/11/10_banques_dimages_gratuites_libres_de_droits-300x169.jpg"
        />,
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('in document', () => {
    render(
      <MediaImage
        src="https://bibliotheques.csdm.qc.ca/files/2018/11/10_banques_dimages_gratuites_libres_de_droits-300x169.jpg"
      />,
    );
    const search = screen.getByTestId('image');
    expect(search).toBeInTheDocument();
  });
});
