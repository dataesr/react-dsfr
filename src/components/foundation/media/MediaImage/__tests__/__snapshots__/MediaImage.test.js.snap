// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MediaImage /> renders correctly 1`] = `
<figure
  className="fr-content-media fr-content-media--md"
  data-testid="image"
>
  <div
    className="fr-content-media__img"
  >
    <img
      alt=""
      src="https://dummyimage.com/300x170/000/fff.png"
    />
  </div>
  <figcaption
    className="fr-content-media__caption"
  />
</figure>
`;
